pylint/checkers/classes/class_checker.py

```diff
@@ -702,17 +711,19 @@ def _has_same_layout_slots(slots, assigned_value):
 }
 
 
-def _scope_default():
-    return collections.defaultdict(list)
+def _scope_default() -> defaultdict[str, list[_AccessNodes]]:
+    return defaultdict(list)
```

Comment:
Should we remove this function ?

Reply:
We can't. Apparently you can't nest `defaultdicts`... Which is probably why this was done like this in the first place.

Reply:
All right ! 

```suggestion
def _scope_default() -> defaultdict[str, list[_AccessNodes]]:
    # It's impossible to nest defaultdicts so we must use a function
    return defaultdict(list)
```